$schema: http://json-schema.org/draft-07/schema#
title: Topological Map Schema
type: object
required:
  - meta
  - metric_map
  - name
  - nodes
  - pointset
additionalProperties: true
properties:
  meta:
    type: object
    properties:
      last_updated:
        type: string
        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{2}-[0-9]{2}-[0-9]{2}$
    required:
      - last_updated
    additionalProperties: true
  metric_map:
    type: string
  name:
    type: string
  nodes:
    type: array
    items:
      type: object
      required:
        - meta
        - node
      properties:
        meta:
          type: object
          required:
            - map
            - node
            - pointset
          properties:
            map:
              type: string
            node:
              type: string
            pointset:
              type: string
        node:
          type: object
          required:
            - edges
            - name
            - parent_frame
          additionalProperties: true
          properties:
            edges:
              type: array
              items:
                type: object
                additionalProperties: true
                required:
                  - edge_id
                  - node
                properties:
                  edge_id:
                    type: string
                  node:
                    type: string
            name:
              type: string
            parent_frame:
              type: string
  pointset:
    type: string
